src/Entity/Product.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace App\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity(repositoryClass="App\Repository\ProductRepository")
+ */
+class Product
+{
+    /**
+     * @ORM\Id()
+     * @ORM\GeneratedValue()
+     * @ORM\Column(type="integer")
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     */
+    private $name;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     */
+    private $description;
+
+    /**
+     * @ORM\Column(type="datetime", length=255)
+     */
+    private $date_of_creation;
+
+    /**
+     * @ORM\Column(type="datetime", length=255)
+     */
+    private $date_of_last_modification;
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    public function setName(string $name): self
+    {
+        $this->name = $name;
+
+        return $this;
+    }
+
+    public function getDescription(): ?string
+    {
+        return $this->description;
+    }
+
+    public function setDescription(string $description): self
+    {
+        $this->description = $description;
+
+        return $this;
+    }
+
+    public function getDateOfCreation(): ?string
+    {
+        return $this->date_of_creation;
+    }
+
+    public function setDateOfCreation(string $date_of_creation): self
+    {
+        $this->date_of_creation = $date_of_creation;
+
+        return $this;
+    }
+
+    public function getDateOfLastModification(): ?string
```

Comment:
incorrect return type, now when $date_of_creation is datetime its not a string but it should be \DateTimeInterface